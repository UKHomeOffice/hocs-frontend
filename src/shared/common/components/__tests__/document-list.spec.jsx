import React from 'react';
import DocumentList from '../document-list.jsx';

describe('Document list component', () => {

    const documentList = [
        ['group 1', [
            { label: 'TEST_DOCUMENT_1', value: 'MOCK_DOC_ID_1', status: 'UPLOADED', hasPdf: true, hasOriginalFile: true },
            { label: 'TEST_DOCUMENT_2', value: 'MOCK_DOC_ID_2', status: 'FAILED_CONVERSION', tags: ['Failed Conversion'], hasPdf: false, hasOriginalFile: true },
            { label: 'TEST_DOCUMENT_3', value: 'MOCK_DOC_ID_3', status: 'FAILED_VIRUS', tags: ['Failed Virus'], hasPdf: false, hasOriginalFile: false },
            { label: 'TEST_DOCUMENT_4', value: 'MOCK_DOC_ID_4', status: 'PENDING', tags: [], hasPdf: false, hasOriginalFile: true },

        ]], ['group 2', [
            { label: 'TEST_DOCUMENT_5', value: 'MOCK_DOC_ID_5', status: 'UPLOADED', hasPdf: true, hasOriginalFile: true },
            { label: 'TEST_DOCUMENT_6', value: 'MOCK_DOC_ID_6', status: 'FAILED_CONVERSION', tags: ['Failed Conversion'], hasPdf: false, hasOriginalFile: true },
            { label: 'TEST_DOCUMENT_7', value: 'MOCK_DOC_ID_7', status: 'FAILED_VIRUS', tags: ['Failed Virus'], hasPdf: false, hasOriginalFile: false },
            { label: 'TEST_DOCUMENT_8', value: 'MOCK_DOC_ID_8', status: 'PENDING', tags: ['Pending'], hasPdf: false, hasOriginalFile: true },
        ]], ['group 3', [
            { label: 'TEST_DOCUMENT_9', value: 'MOCK_DOC_ID_9', status: 'UPLOADED', hasPdf: true, hasOriginalFile: true },
            { label: 'TEST_DOCUMENT_10', value: 'MOCK_DOC_ID_10', status: 'FAILED_CONVERSION', tags: ['Failed Conversion'], hasPdf: false, hasOriginalFile: true },
            { label: 'TEST_DOCUMENT_11', value: 'MOCK_DOC_ID_11', status: 'FAILED_VIRUS',  tags: ['Failed Virus'], hasPdf: false, hasOriginalFile: false },
            { label: 'TEST_DOCUMENT_12', value: 'MOCK_DOC_ID_12', status: 'PENDING', tags: ['UNKNOWN', 'Pending'], hasPdf: false, hasOriginalFile: true },
            { label: 'TEST_DOCUMENT_13', value: 'MOCK_DOC_ID_13', status: 'INVALID', tags: ['INVALID'], hasPdf: false, hasOriginalFile: false },
        ]]
    ];

    it('should render with default props', () => {
        const wrapper = render(<DocumentList caseId={'MOCK_CASE_ID'} clickHandler={() => { }} />);
        expect(wrapper).toBeDefined();
        expect(wrapper).toMatchSnapshot();
    });

    it('should render with a document list when provided in props', () => {
        const wrapper = render(<DocumentList caseId={'MOCK_CASE_ID'} clickHandler={() => { }} documents={documentList} />);
        expect(wrapper).toBeDefined();
        expect(wrapper).toMatchSnapshot();
    });

    it('should support a click handler when passed in props', () => {
        const mockClickHandler = jest.fn();
        const wrapper = shallow(<DocumentList caseId={'MOCK_CASE_ID'} clickHandler={mockClickHandler} documents={documentList} />);
        expect(wrapper).toBeDefined();
        wrapper.find('#MOCK_DOC_ID_1-pdf').simulate('click', { preventDefault: jest.fn() });
        expect(mockClickHandler).toHaveBeenCalled();
    });

    it('should display None when there are no documents in a group', () => {
        const emptyDocumentList = [
            ['group 1'], ['group 2', []]
        ];

        const wrapper = render(<DocumentList caseId={'MOCK_CASE_ID'} clickHandler={() => { }} documents={emptyDocumentList} />);
        expect(wrapper).toBeDefined();
        expect(wrapper).toMatchSnapshot();
        const cells = wrapper.find('td');
        expect(cells).toHaveLength(2);
        expect(cells.first().text()).toBe('None');
        expect(cells.last().text()).toBe('None');
    });
});
