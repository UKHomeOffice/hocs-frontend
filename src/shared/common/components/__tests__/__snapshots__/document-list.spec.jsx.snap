// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Document list component should render with a document list when provided in props 1`] = `
<table
  class="govuk-table"
>
  <caption
    class="govuk-table__caption"
  >
    Documents
  </caption>
  <tbody
    class="govuk-table__body"
  >
    <tr
      class="govuk-table__row"
    >
      <td
        class="govuk-table__cell"
      >
        <strong
          class="govuk-tag margin-right--small"
        >
          ORIGINAL
        </strong>
        <strong
          class="govuk-tag"
        />
      </td>
      <td
        class="govuk-table__cell"
      />
      <td
        class="govuk-table__cell"
      >
        <a
          class="govuk-link"
          download="MOCK_DOC_ID_1"
          href="/case/MOCK_CASE_ID/stage/undefined/download/document/MOCK_DOC_ID_1/pdf"
          id="MOCK_DOC_ID_1-pdf"
        >
          Preview
        </a>
      </td>
      <td
        class="govuk-table__cell"
      >
        <a
          class="govuk-link"
          href="/case/MOCK_CASE_ID/stage/undefined/download/document/MOCK_DOC_ID_1/original"
        >
          Download
        </a>
      </td>
    </tr>
    <tr
      class="govuk-table__row"
    >
      <td
        class="govuk-table__cell"
      >
        <strong
          class="govuk-tag margin-right--small"
        >
          ORIGINAL
        </strong>
        <strong
          class="govuk-tag"
        />
      </td>
      <td
        class="govuk-table__cell"
      />
      <td
        class="govuk-table__cell"
      >
        <a
          class="govuk-link"
          download="MOCK_DOC_ID_2"
          href="/case/MOCK_CASE_ID/stage/undefined/download/document/MOCK_DOC_ID_2/pdf"
          id="MOCK_DOC_ID_2-pdf"
        >
          Preview
        </a>
      </td>
      <td
        class="govuk-table__cell"
      >
        <a
          class="govuk-link"
          href="/case/MOCK_CASE_ID/stage/undefined/download/document/MOCK_DOC_ID_2/original"
        >
          Download
        </a>
      </td>
    </tr>
    <tr
      class="govuk-table__row"
    >
      <td
        class="govuk-table__cell"
      >
        <strong
          class="govuk-tag margin-right--small"
        >
          ORIGINAL
        </strong>
        <strong
          class="govuk-tag"
        />
      </td>
      <td
        class="govuk-table__cell"
      />
      <td
        class="govuk-table__cell"
      >
        <a
          class="govuk-link"
          download="MOCK_DOC_ID_3"
          href="/case/MOCK_CASE_ID/stage/undefined/download/document/MOCK_DOC_ID_3/pdf"
          id="MOCK_DOC_ID_3-pdf"
        >
          Preview
        </a>
      </td>
      <td
        class="govuk-table__cell"
      >
        <a
          class="govuk-link"
          href="/case/MOCK_CASE_ID/stage/undefined/download/document/MOCK_DOC_ID_3/original"
        >
          Download
        </a>
      </td>
    </tr>
  </tbody>
</table>
`;

exports[`Document list component should render with default props 1`] = `
<table
  class="govuk-table"
>
  <caption
    class="govuk-table__caption"
  >
    Documents
  </caption>
  <tbody
    class="govuk-table__body"
  />
</table>
`;
